secret.go

```diff
@@ -139,21 +139,23 @@ func (c *Client) ListSecrets() ([]string, error) {
 
 // GetAllSecrets gets all secret objects with their values in a configured application
 func (c *Client) GetAllSecrets() ([]Secret, error) {
```

Comment:
I do like this addition though! Rather than looping through to fetch the secrets, it would be a lot more efficient to pull them all at once with the [`​Open​App​Secrets`](https://developer.hashicorp.com/hcp/api-docs/vault-secrets#OpenAppSecrets) call.